bottlerocket-os/bottlerocket
# update grub to 2.06 and implement hybrid boot support for x86_64

**Issue number:**
Fixes #1697


**Description of changes:**
Update grub to 2.06, from Amazon Linux by way of Fedora. Build both EFI and BIOS images for GRUB.

Adjust default grub configurations to take advantage of the improved console support on EFI and BIOS, and to disable module loading for security reasons.

Implement hybrid support for x86_64 images, so they work with either BIOS or EFI firmware.


**Testing done:**
Verified that the GPT priority functionality works correctly by following these steps for x86_64 (BIOS, EFI) and aarch64 (EFI):
* launch a test 1.2.0 image
* update to a test 1.3.0 image
* verify that 1.3.0 comes up, reboot, and confirm it's still 1.3.0
* downgrade to a test 1.2.0 image
* verify that 1.2.0 comes up, reboot, and confirm it's still 1.2.0
* update to a defective 1.4.0 image
* verify that the boot fails, reboot, and confirm that it's 1.2.0

Verified that the grub console output was visible on these platforms:
* EC2 virtualized, for x86_64 (BIOS, EFI) and aarch64 (EFI)
* EC2 metal, for x86_64 (BIOS) and aarch64 (EFI)
* VMware guest, for x86_64 (BIOS)

Output was not visible for the VMware guest under EFI, although the guest booted OK. I don't consider this a blocker since we default to BIOS on that platform.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.
